### PR TITLE
bpo-32603: Deprecation warning on strings used in re module

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue32603

### CODE REVIEW

File: Doc/library/re.rst [hunk 1]
```diff
@@ -162,11 +162,11 @@ The special characters are:
 
    If you're not using a raw string to express the pattern, remember that Python
    also uses the backslash as an escape sequence in string literals; if the escape
-   sequence isn't recognized by Python's parser, the backslash and subsequent
-   character are included in the resulting string.  However, if Python would
-   recognize the resulting sequence, the backslash should be repeated twice.  This
-   is complicated and hard to understand, so it's highly recommended that you use
-   raw strings for all but the simplest expressions.
+   sequence isn't recognized by Python's parser, it will generate a
+   :exc:`DeprecationWarning` and in the future this will become a :exc:`Syntaxerror`.
```
Comment (csabella) [inline Doc/library/re.rst]: Syntaxerror -> SyntaxError

**Comment by csabella** [comment]
IMHO, I actually thought the warning about the DeprecationWarning and SyntaxError should come in one of the introduction paragraphs since the raw strings are discussed there.

In the paragraph that starts with, `Regular expressions use the backslash character`, maybe a line at the end with something like 'Also, please note that any invalid escape sequences in Python's usage of the backslash now generate a DeprecationWarning and, in the future, a SyntaxError.  This will happen even if it is a valid escape sequence for a regular expression.'

**Comment by pablogsal** [comment]
Thank you very much @csabella. I have added some lines as you suggested in the introductory paragraphs.

File: Doc/library/re.rst [hunk 2]
```diff
@@ -28,7 +28,10 @@ character for the same purpose in string literals; for example, to match
 a literal backslash, one might have to write ``'\\\\'`` as the pattern
 string, because the regular expression must be ``\\``, and each
 backslash must be expressed as ``\\`` inside a regular Python string
-literal.
+literal. Also, please note that any invalid escape sequences in Python's
+usage of the backslash in string literals now generate a DeprecationWarning
```
Comment (matrixise) [inline Doc/library/re.rst]: Could you use a link to the DeprecationWarning exception like that 
```
:exc:`DeprecationWarning`
```

same for the SyntaxError exception

**Review by matrixise — CHANGES_REQUESTED** [review]

**Comment by pablogsal** [comment]
CC: @JulienPalard

**Pull request merged**